components/scream/src/share/field/field_layout.cpp

```diff
@@ -103,36 +101,36 @@ LayoutType get_layout_type (const std::vector<FieldTag>& field_tags) {
       result = LayoutType::Scalar2D;
       break;
     case 1:
-      // The only tag left should be 'CMP', 'TL', 'VAR', or 'LEV'/'ILEV'
-      if (tags[0]==CMP || tags[0]==TL || tags[0]==VAR) {
+      // The only tag left should be 'CMP', 'TL', or 'LEV'/'ILEV'
+      if (tags[0]==CMP || tags[0]==TL) {
         result = LayoutType::Vector2D;
       } else if (tags[0]==LEV || tags[0]==ILEV) {
         result = LayoutType::Scalar3D;
       }
       break;
     case 2:
       // Possible scenarios:
-      //  1) <CMP|VAR|TL,LEV|ILEV>
-      //  2) <VAR|TL,CMP>
-      //  3) <TL,VAR>
+      //  1) <CMP|TL,LEV|ILEV>
+      //  2) <TL,CMP>
+      //  3) <TL>
       //  4) <CMP1,CMP2>
-      if ( (tags[1]==LEV || tags[1]==ILEV) && (tags[0]==CMP || tags[0]==VAR || tags[0]==TL)) {
+      if ( (tags[1]==LEV || tags[1]==ILEV) && (tags[0]==CMP || tags[0]==TL)) {
         result = LayoutType::Vector3D;
       } else if ((tags[0]==CMP1 && tags[1]==CMP2) ||
-                 (tags[1]==CMP && (tags[0]==VAR || tags[0]==TL)) ||
-                 (tags[0]==TL && tags[1]==VAR)) {
+                 (tags[1]==CMP && tags[0]==TL) ||
+                  tags[0]==TL) {
```

Comment:
To be a 2d tensor, there should be 2 tags left, so tags[0]==TL alone is wrong.

As stated above, option 3 goes away, so this last condition should be removed.